components/user-store/org.wso2.carbon.identity.user.store.configuration/src/main/java/org/wso2/carbon/identity/user/store/configuration/utils/AttributeMappingBuilder.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.com).
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.user.store.configuration.utils;
+
+import org.wso2.carbon.identity.user.store.configuration.internal.UserStoreConfigListenersHolder;
+import org.wso2.carbon.identity.user.store.configuration.model.UserStoreAttributeMappings;
+
+/**
+ * Build attribute mappings.
+ */
+public class AttributeMappingBuilder {
+
+    private AttributeMappingBuilder() {
+
+    }
+
+    private static final class BuilderHolder {
+
+        static final AttributeMappingBuilder BUILDER = new AttributeMappingBuilder();
+    }
+
+    public static AttributeMappingBuilder getInstance() {
+
+        return BuilderHolder.BUILDER;
+    }
+
+    /**
+     * Build attribute mappings for user store types and store in {@link UserStoreConfigListenersHolder}.
+     */
+    public void build() {
```

Comment:
It would be nice to return the built information so that the caller can handle it accordingly.

Reply:
What is meant by built information? A boolean value to indicate success or failed?

Reply:
No, I'm referring to the `UserStoreAttributeMappings` built from the method. If you see the caller's perspective,

```    
public UserStoreAttributeMappings getUserStoreAttributeMappings() {

        if (userStoreAttributeMappings == null) {
            // Read user store attributes mappings from files.
            AttributeMappingBuilder.getInstance().build();
        }
        return userStoreAttributeMappings;
}
```

Calling this `build()` method does make sense. If you return the `UserStoreAttributeMappings` and caller decides how to keep them for later usages, that would make this more readable.

Reply:
We removed this class and move this logic to component class. fixed in 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

Reply:
We removed this class and moved the logic to component class. fixed in https://github.com/wso2/carbon-identity-framework/pull/3781